src/index.js

```diff
@@ -114,46 +137,50 @@ async function fetchWithNetworkEvents(el: Element, url: string, options: Request
 }
 
 async function check(autoCheckElement: AutoCheckElement) {
-  const src = autoCheckElement.src
-  if (!src) {
-    throw new Error('missing src')
-  }
-  const csrf = autoCheckElement.csrf
-  if (!csrf) {
-    throw new Error('missing csrf')
-  }
   const input = autoCheckElement.input
-  if (!input) return
+  if (!input) {
+    return
+  }
 
+  const src = autoCheckElement.src
+  const csrf = autoCheckElement.csrf
   const state = states.get(autoCheckElement)
-  if (!state) return
+
+  // If some attributes are missing we want to exit early and make sure that the element is valid.
+  if (!src || !csrf || !state) {
+    if (autoCheckElement.required) {
+      input.setCustomValidity('')
+    }
+    input.dispatchEvent(new CustomEvent('auto-check-complete', {bubbles: true}))
```

Comment:
These attributes are prerequisites to the element's behavior. If they are not present, no events will be dispatched and the element is inert.
```suggestion
```